TombEngine/Objects/Generic/Object/Pushables/Context.h

```diff
@@ -4,13 +4,13 @@ class Vector3i;
 
 namespace TEN::Entities::Generic
 {
-	enum class PushableEnvironmentState
+	enum class PushableEnvironmentType
 	{
 		Air,
-		Ground,
-		Slope,
-		Water,
-		GroundWater
+		FlatFloor,
+		SlopedFloor,
+		DeepWater,
+		ShallowWater
```

Comment:
I think these names:

DeepWater and ShallowWater are unsuitable, as may lead to a concepts confusion.

Usually I understand DeepWater to the water rooms that have a depth of 3 clicks or more. And Shallow Water to the ones that have a depth of 2 clicks or less (usually used to identify the parts where Lara can swim and where Lara can wade).

However this context function is not checking that. What it needs to see is if the block is in the water room floor, or is higher (so is sinking or floating).

Is something similar like the flat floor or on air states, but for water rooms. ( their own version because are very frequent cases so I could use those types instead of check is on the ground, Is it also in water room?, so the GroundWater state implicit both conditions of that situation.

Formally, I also was using a ShallowWater type so I had 3, DeepWater, ShallowWater and GroundWater) but reviewing  the code, I found that the behvarious that differ DeepWater from ShallowWater were mostly anecdotal, so I combined those cases into only one (Water), and did the distinction in the states functions (that were only in a couple of moments).